src/Components/Main/Pages/ProfilePage/ProfileInfo/ProfileInfoContainer.js

```diff
@@ -1,33 +1,43 @@
 import { ProfileInfo } from './ProfileInfo';
 import { useCallback, useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { useParams } from 'react-router-dom';
 import { useMounted } from '../../../../../hooks/useMounted';
 import { getProfile, getStatus, updateStatus } from '../../../../../redux/thunks/profile';
+import { isFetchingProfileSelector, isFetchingStatusSelector } from '../../../../../redux/selectors/loading';
+import { profileSelector, statusSelector } from '../../../../../redux/selectors/profile';
+import { idSelector } from '../../../../../redux/selectors/auth';
+import { useNavigate, useParams } from 'react-router-dom';
 
 export const ProfileInfoContainer = () => {
-  const profile = useSelector((state) => state.profile.profile);
-  const isFetchingProfile = useSelector((state) => state.profile.isFetchingProfile);
-  const status = useSelector((state) => state.profile.status);
-  const isFetchingStatus = useSelector((state) => state.profile.isFetchingStatus);
+  const profile = useSelector(profileSelector);
+  const isFetchingProfile = useSelector(isFetchingProfileSelector);
+  const status = useSelector(statusSelector);
+  const isFetchingStatus = useSelector(isFetchingStatusSelector);
+  const authorizedUserId = useSelector(idSelector);
 
   const dispatch = useDispatch();
   const updateStatusCallback = useCallback(
     (status) => dispatch(updateStatus(status)),
     [dispatch]
   );
 
-  let {id} = useParams();
-
-  const mounted = useMounted();
+  let {id: paramId} = useParams();
```

Comment:
You could try the improvement:
```suggestion
  let {id: paramId = authorizedUserId} = useParams();
```

And in code just:
```
if (paramId) {
      dispatch(getProfile(paramId));
      dispatch(getStatus(paramId));
    } else {
      navigate('/login');
    }
```

Reply:
Done!